### PR TITLE
`r/aws_rds_cluster`: Conflict `snapshot_identifier` and `global_cluster_identifier`

### LINKED ISSUE #29187: [Bug]: Cannot create Global Aurora (MySQL) cluster based upon snapshot
### Terraform Core Version

1.3.7

### AWS Provider Version

4.52.0

### Affected Resource(s)

* `aws_db_cluster_snapshot` (data)
* `aws_rds_global_cluster`
* `aws_rds_cluster`
* `aws_rds_cluster_instance`

### Expected Behavior

A MySQL Global Aurora cluster should be created with a primary cluster and writer instance based upon a snapshot, it should look something like this:

Screenshot 2023-01-31 at 16-37-35 RDS Management Console

Note that this correct looking cluster was created using the exact same code just with the `snapshot_identifier = data.aws_db_cluster_snapshot.sre_5136_to_be_upgraded.id` part commented out.

### Actual Behavior

A MySQL Global Aurora cluster is created but the primary cluster and instance are not added to it, they are created separately as a standalone regional cluster.

Screenshot 2023-01-31 at 14-07-43 RDS Management Console

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```terraform
# Create Initial Cluster to be Copied Next
# -----------------------------------------------------------------------------

resource "aws_rds_global_cluster" "sre_5136" {
  global_cluster_identifier = "sre-5136-global-test"
  engine                    = "aurora"
  engine_version            = "5.6.mysql_aurora.1.22.4"
  storage_encrypted         = true
}

resource "aws_rds_cluster" "sre_5136_primary" {
  engine                    = aws_rds_global_cluster.sre_5136.engine
  engine_version            = aws_rds_global_cluster.sre_5136.engine_version
  cluster_identifier        = "sre-5136-test-primary-cluster"
  master_username           = "example_user"
  master_password           = "password"
  database_name             = "example_db"
  global_cluster_identifier = aws_rds_global_cluster.sre_5136.id
  db_subnet_group_name      = "default"
}

resource "aws_rds_cluster_instance" "sre_5136_primary" {
  engine               = aws_rds_global_cluster.sre_5136.engine
  engine_version       = aws_rds_global_cluster.sre_5136.engine_version
  identifier           = "sre-5136-test-primary-cluster-instance"
  cluster_identifier   = aws_rds_cluster.sre_5136_primary.id
  instance_class       = "db.r4.large"
  db_subnet_group_name = "default"
}


# Try Creating a New Upgraded Cluster Based on Snapshot From Above Cluster
# -----------------------------------------------------------------------------

# Grab the most recent snapshot from the RDS cluster above.
data "aws_db_cluster_snapshot" "sre_5136_to_be_upgraded" {
  db_cluster_identifier = aws_rds_cluster.sre_5136_primary.id
  most_recent           = true
}

# Upgrade to the latest engine version supported by the snapshot.
resource "aws_rds_global_cluster" "sre_5136_upgraded" {
  global_cluster_identifier = "sre-5136-global-test-upgraded"
  engine                    = "aurora"
  engine_version            = "5.6.mysql_aurora.1.22.4"
  storage_encrypted         = true
}

# Use the snapshot of the above database to create a new dev database.
resource "aws_rds_cluster" "sre_5136_upgraded" {
  engine                    = aws_rds_global_cluster.sre_5136_upgraded.engine
  engine_version            = aws_rds_global_cluster.sre_5136_upgraded.engine_version
  cluster_identifier        = "sre-5136-upgraded-cluster"
  
  # Commented out as not needed when using a snapshot but needing when testing creation not based on a snapshot.
  # master_username           = "example_user"
  # master_password           = "password"
  # database_name             = "example_db"
  
  global_cluster_identifier = aws_rds_global_cluster.sre_5136_upgraded.id
  db_subnet_group_name      = "default"

  snapshot_identifier = data.aws_db_cluster_snapshot.sre_5136_to_be_upgraded.id

  lifecycle {
    ignore_changes = [snapshot_identifier]
  }
}

resource "aws_rds_cluster_instance" "sre_5136_upgraded" {
  engine               = aws_rds_global_cluster.sre_5136_upgraded.engine
  engine_version       = aws_rds_global_cluster.sre_5136_upgraded.engine_version
  identifier           = "sre-5136-upgraded-test-primary-cluster-instance"
  cluster_identifier   = aws_rds_cluster.sre_5136_upgraded.id
  instance_class       = "db.r5.large"
  db_subnet_group_name = "default"
}
```

### Steps to Reproduce

1.  Create the first Global Aurora and snapshot it and create a manual snaphshot for testing.
2. Add the second Global Aurora and `terraform apply`.

I was then double checking what **should** happen by creating the second cluster from scratch by uncommenting the `master_username`, `master_password`, and `database_name` and commenting out this section:

```terraform
  snapshot_identifier = data.aws_db_cluster_snapshot.sre_5136_to_be_upgraded.id

  lifecycle {
    ignore_changes = [snapshot_identifier]
  }
```

The Global cluster is built correctly then.

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

* Old issue:  https://github.com/hashicorp/terraform-provider-aws/issues/9210
* Docs:  https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/db_cluster_snapshot

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by jar-b**
### Description
Triggers a conflict when both `snapshot_identifier` and `global_cluster_identifier` are provided in the same configuration.  It is not currently possible to restore from snapshot AND join an existing global cluster in a single operation. This plan time check prevents configurations which will succeed with unexpected results (see #29187).

### Relations
Closes #29187 

### References
- https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-global-database-getting-started.html#aurora-global-database.use-snapshot
- See https://github.com/hashicorp/terraform-provider-aws/issues/29187#issuecomment-1477931892 for an example of restoring a global cluster from snapshot with terraform

### Output from Acceptance Testing
**NOTE:** Failing tests do not appear related to this change.

```console
$ make testacc PKG=rds TESTS=TestAccRDSCluster_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/rds/... -v -count 1 -parallel 20 -run='TestAccRDSCluster_'  -timeout 180m
--- PASS: TestAccRDSCluster_basic (144.20s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_encryptedRestore
--- PASS: TestAccRDSCluster_EngineMode_global (151.79s)
=== CONT  TestAccRDSCluster_serverlessV2ScalingConfiguration
--- PASS: TestAccRDSCluster_encrypted (151.80s)
=== CONT  TestAccRDSCluster_scaling
--- PASS: TestAccRDSCluster_iamAuth (152.21s)
=== CONT  TestAccRDSCluster_SnapshotIdentifierVPCSecurityGroupIDs_tags
--- PASS: TestAccRDSCluster_GlobalClusterIdentifierEngineMode_provisioned (174.67s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_vpcSecurityGroupIDs
--- PASS: TestAccRDSCluster_kmsKey (187.35s)
=== CONT  TestAccRDSCluster_port
--- PASS: TestAccRDSCluster_EngineMode_multiMaster (189.50s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_tags
--- PASS: TestAccRDSCluster_backupsUpdate (189.80s)
=== CONT  TestAccRDSCluster_GlobalClusterIdentifier_secondaryClustersWriteForwarding
--- PASS: TestAccRDSCluster_deletionProtection (191.87s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_preferredMaintenanceWindow
=== NAME  TestAccRDSCluster_port
    cluster_test.go:1471: Step 1/2 error: Error running apply: exit status 1

        Error: creating RDS Cluster (tf-acc-test-8674226860890088684): InvalidParameterCombination: The Parameter Group default.aurora-postgresql13 with DBParameterGroupFamily aurora-postgresql13 cannot be u
sed for this instance. Please use a Parameter Group with DBParameterGroupFamily aurora-postgresql14
                status code: 400, request id: 0ed4fa2d-c656-459e-84d5-0aacb94ef618

          with aws_rds_cluster.test,
          on terraform_plugin_test.tf line 2, in resource "aws_rds_cluster" "test":
           2: resource "aws_rds_cluster" "test" {

--- FAIL: TestAccRDSCluster_port (5.46s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_preferredBackupWindow
=== NAME  TestAccRDSCluster_GlobalClusterIdentifier_secondaryClustersWriteForwarding
    cluster_test.go:1443: Step 1/1 error: Error running apply: exit status 1

        Error: creating RDS Global Cluster: InvalidParameterValue: The requested engine version was not found or does not support global functionality
                status code: 400, request id: 50b81a05-caa4-4f1a-bda5-0f6a4ba67447

          with aws_rds_global_cluster.test,
          on terraform_plugin_test.tf line 18, in resource "aws_rds_global_cluster" "test":
          18: resource "aws_rds_global_cluster" "test" {

--- FAIL: TestAccRDSCluster_GlobalClusterIdentifier_secondaryClustersWriteForwarding (9.54s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_masterUsername
--- PASS: TestAccRDSCluster_updateIAMRoles (223.18s)
=== CONT  TestAccRDSCluster_GlobalClusterIdentifier_replicationSourceIdentifier
--- PASS: TestAccRDSCluster_EnabledCloudWatchLogsExports_mySQL (231.50s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_masterPassword
--- PASS: TestAccRDSCluster_networkType (276.51s)
=== CONT  TestAccRDSCluster_GlobalClusterIdentifier_primarySecondaryClusters
--- PASS: TestAccRDSCluster_Scaling_defaultMinCapacity (284.81s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_kmsKeyID
--- PASS: TestAccRDSCluster_copyTagsToSnapshot (290.75s)
=== CONT  TestAccRDSCluster_missingUserNameCausesError
--- PASS: TestAccRDSCluster_missingUserNameCausesError (5.00s)
=== CONT  TestAccRDSCluster_SnapshotIdentifierEngineVersion_equal
--- PASS: TestAccRDSCluster_serverlessV2ScalingConfiguration (199.13s)
=== CONT  TestAccRDSCluster_SnapshotIdentifierEngineVersion_different
--- PASS: TestAccRDSCluster_enableHTTPEndpoint (353.31s)
=== CONT  TestAccRDSCluster_availabilityZones
--- PASS: TestAccRDSCluster_PointInTimeRestore_enabledCloudWatchLogsExports (446.19s)
=== CONT  TestAccRDSCluster_SnapshotIdentifierEngineMode_provisioned
--- PASS: TestAccRDSCluster_engineMode (446.51s)
=== CONT  TestAccRDSCluster_pointInTimeRestore
--- PASS: TestAccRDSCluster_scaling (306.70s)
=== CONT  TestAccRDSCluster_SnapshotIdentifierEngineMode_parallelQuery
    cluster_test.go:1698: Step 1/1 error: Error running apply: exit status 1

        Error: creating RDS Cluster (tf-acc-test-810470511273405909-source): InvalidParameterCombination: Cannot find version null for aurora
                status code: 400, request id: 5a337cd3-958e-40b7-b2ec-246ec2786b13

          with aws_rds_cluster.source,
          on terraform_plugin_test.tf line 2, in resource "aws_rds_cluster" "source":
           2: resource "aws_rds_cluster" "source" {

--- FAIL: TestAccRDSCluster_SnapshotIdentifierEngineMode_parallelQuery (4.52s)
=== CONT  TestAccRDSCluster_dbSubnetGroupName
--- PASS: TestAccRDSCluster_snapshotIdentifier (474.00s)
=== CONT  TestAccRDSCluster_SnapshotIdentifier_deletionProtection
--- PASS: TestAccRDSCluster_availabilityZones (145.86s)
=== CONT  TestAccRDSCluster_backtrackWindow
--- PASS: TestAccRDSCluster_SnapshotIdentifier_encryptedRestore (426.04s)
=== CONT  TestAccRDSCluster_dbClusterInstanceClass
--- PASS: TestAccRDSCluster_SnapshotIdentifierVPCSecurityGroupIDs_tags (460.79s)
=== CONT  TestAccRDSCluster_takeFinalSnapshot
--- PASS: TestAccRDSCluster_dbSubnetGroupName (153.30s)
=== CONT  TestAccRDSCluster_iops
--- PASS: TestAccRDSCluster_SnapshotIdentifier_preferredMaintenanceWindow (424.97s)
=== CONT  TestAccRDSCluster_allocatedStorage
--- PASS: TestAccRDSCluster_SnapshotIdentifier_masterUsername (425.31s)
=== CONT  TestAccRDSCluster_allowMajorVersionUpgrade
--- PASS: TestAccRDSCluster_SnapshotIdentifier_vpcSecurityGroupIDs (470.06s)
=== CONT  TestAccRDSCluster_onlyMajorVersion
--- PASS: TestAccRDSCluster_SnapshotIdentifier_tags (465.34s)
=== CONT  TestAccRDSCluster_storageType
--- PASS: TestAccRDSCluster_SnapshotIdentifier_preferredBackupWindow (465.65s)
=== CONT  TestAccRDSCluster_allowMajorVersionUpgradeWithCustomParameters
--- PASS: TestAccRDSCluster_SnapshotIdentifier_masterPassword (435.69s)
=== CONT  TestAccRDSCluster_GlobalClusterIdentifierEngineMode_global
--- PASS: TestAccRDSCluster_backtrackWindow (183.75s)
=== CONT  TestAccRDSCluster_allowMajorVersionUpgradeWithCustomParametersApplyImm
--- PASS: TestAccRDSCluster_SnapshotIdentifier_kmsKeyID (466.41s)
=== CONT  TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_remove
--- PASS: TestAccRDSCluster_SnapshotIdentifierEngineVersion_equal (478.94s)
=== CONT  TestAccRDSCluster_identifierGenerated
--- PASS: TestAccRDSCluster_GlobalClusterIdentifierEngineMode_global (169.53s)
=== CONT  TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_add
--- PASS: TestAccRDSCluster_SnapshotIdentifierEngineVersion_different (489.36s)
=== CONT  TestAccRDSCluster_identifierPrefix
--- PASS: TestAccRDSCluster_SnapshotIdentifierEngineMode_provisioned (415.55s)
=== CONT  TestAccRDSCluster_engineVersion
--- PASS: TestAccRDSCluster_pointInTimeRestore (418.59s)
=== CONT  TestAccRDSCluster_EngineMode_parallelQuery
--- PASS: TestAccRDSCluster_takeFinalSnapshot (256.16s)
=== CONT  TestAccRDSCluster_engineVersionWithPrimaryInstance
=== NAME  TestAccRDSCluster_EngineMode_parallelQuery
    cluster_test.go:1125: Step 1/1 error: Error running apply: exit status 1

        Error: creating RDS Cluster (tf-acc-test-1459281107440858057): InvalidParameterCombination: Cannot find version null for aurora
                status code: 400, request id: 10dc314c-ea60-4fc2-ab41-fd9e468715b5

          with aws_rds_cluster.test,
          on terraform_plugin_test.tf line 2, in resource "aws_rds_cluster" "test":
           2: resource "aws_rds_cluster" "test" {

--- FAIL: TestAccRDSCluster_EngineMode_parallelQuery (4.51s)
=== CONT  TestAccRDSCluster_tags
--- PASS: TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_remove (168.02s)
=== CONT  TestAccRDSCluster_disappears
--- PASS: TestAccRDSCluster_identifierGenerated (184.77s)
=== CONT  TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_update
--- PASS: TestAccRDSCluster_identifierPrefix (133.64s)
--- PASS: TestAccRDSCluster_SnapshotIdentifier_deletionProtection (527.39s)
--- PASS: TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_add (173.25s)
--- PASS: TestAccRDSCluster_tags (163.54s)
--- PASS: TestAccRDSCluster_disappears (152.58s)
--- PASS: TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_update (164.11s)
=== NAME  TestAccRDSCluster_allowMajorVersionUpgrade
    cluster_test.go:213: Step 3/3 error: Check failed: Check 4/4 error: aws_rds_cluster.test: Attribute 'engine_version' expected "13.5", got "12.9"
--- PASS: TestAccRDSCluster_engineVersion (473.52s)
--- PASS: TestAccRDSCluster_onlyMajorVersion (963.53s)
--- FAIL: TestAccRDSCluster_allowMajorVersionUpgrade (1248.75s)
--- PASS: TestAccRDSCluster_ReplicationSourceIdentifier_kmsKeyID (2158.29s)
--- PASS: TestAccRDSCluster_dbClusterInstanceClass (1734.17s)
--- PASS: TestAccRDSCluster_engineVersionWithPrimaryInstance (1519.82s)
--- PASS: TestAccRDSCluster_EnabledCloudWatchLogsExports_postgresql (2483.04s)
--- PASS: TestAccRDSCluster_iops (1883.29s)
--- PASS: TestAccRDSCluster_allocatedStorage (1893.50s)
--- PASS: TestAccRDSCluster_storageType (1965.67s)
--- PASS: TestAccRDSCluster_allowMajorVersionUpgradeWithCustomParameters (2144.30s)
--- PASS: TestAccRDSCluster_GlobalClusterIdentifier_replicationSourceIdentifier (2595.85s)
--- PASS: TestAccRDSCluster_allowMajorVersionUpgradeWithCustomParametersApplyImm (2316.59s)
--- PASS: TestAccRDSCluster_GlobalClusterIdentifier_primarySecondaryClusters (2827.92s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/rds        3107.866s
FAIL
make: *** [testacc] Error 1
```

### CODE REVIEW

File: website/docs/r/rds_cluster.html.markdown [hunk 1]
```diff
@@ -191,7 +191,7 @@ The following arguments are supported:
...
-* `snapshot_identifier` - (Optional) Specifies whether or not to create this cluster from a snapshot. You can use either the name or ARN when specifying a DB cluster snapshot, or the ARN when specifying a DB snapshot.
+* `snapshot_identifier` - (Optional) Specifies whether or not to create this cluster from a snapshot. You can use either the name or ARN when specifying a DB cluster snapshot, or the ARN when specifying a DB snapshot. Conflicts with `global_cluster_identifier`. Clusters cannot be restored from snapshot **and** joined to an existing global cluster in a single operation. See the [AWS documentation](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-global-database-getting-started.html#aurora-global-database.use-snapshot) or this [`aws_rds_global_cluster` example](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_global_cluster#new-global-cluster-from-existing-db-cluster) for instructions on building a global cluster starting from a snapshot.
```
Comment (gdavison) [inline website/docs/r/rds_cluster.html.markdown:194]: The Terraform example linked doesn't seem to be an example of creating a cluster from a snapshot. If it's similar, the example should say that it's similar, or else a new example should be added

**Review by gdavison — COMMENTED** [review]
The code change looks good. I got really stuck on the documentation for a couple reasons. First, the linked example was for a different case. Secondly, because the creation order of the RDS global cluster and region cluster is opposite from most other global clusters, such as ElastiCache, I was confused by the example.

File: website/docs/r/rds_cluster.html.markdown (hunk 1 above)
Comment (jar-b) [reply website/docs/r/rds_cluster.html.markdown:194]: Thanks for the feedback - I was on the fence about that Terraform example. It was helpful while researching alternatives for "restoring" global RDS clusters, but I think it's missing too much of the snapshot-related context.

[This comment](https://github.com/hashicorp/terraform-provider-aws/issues/29187#issuecomment-1477931892) on the original issue has a more targeted example, and perhaps should be migrated to this document for a proper example to link here instead.

Comment (gdavison) [reply website/docs/r/rds_cluster.html.markdown:194]: Yes, that looks good

**Pull request merged**